mlflow/langchain/langchain_tracer.py

```diff
@@ -165,25 +139,24 @@ def _end_span(
                 status=status,
             )
 
-    def _is_trace_active(self, request_id: str) -> bool:
-        """Check if a trace with the given request ID is active (i.e. not ended yet)"""
-        return (
-            # Case 1: The root span is started by this callback, the ID
-            # should be in the active list, otherwise it's already ended.
-            request_id in self._active_request_ids
-            # Case 2: The root span is created by fluent API outside this callback.
-            # In this case, we check the context to see if the trace is active or not.
-            or (
-                (active_span := mlflow.get_current_active_span())
-                and (active_span.request_id == request_id)
-            )
-            # Case 3: The root span is created by client API outside this callback,
-            # and passed via the `parent_span` argument of the callback. In this case,
-            # we have no way to check if it is active or not, so just assume it is.
-            or self._parent_span
-        )
+        st = self._run_span_mapping.pop(str(run_id), None)
+        if st and self._set_span_in_context:
+            if st.token is None:
+                raise MlflowException(
+                    f"Token for span {st.span} is not found. Cannot detach the span from context."
+                )
+            detach_span_from_context(st.token)
+
+    def flush(self):
+        """Flush the state of the tracer."""
+        # Ideally, all spans should be popped and ended. However, LangChain sometimes
+        # does not trigger the end event properly and some spans may be left open.
```

Comment:
Do we end those spans?

Reply:
Yes, this is the reason we call `tracer.flush` in the patched prediction function.

Reply:
But here we just detach spans? Is it possible that _end_span is not called

Reply:
Ahh I got what you meant. Hmm we could do that, but that means this function `flush` the backend state. The original intention for this function is just clean up the client-side.

One issue of force-ending spans here is that we cannot set the status/outputs /timestamp correctly, therefore, the resulted spans will look weird anyway. Then I guess it is less confusing to keep the span in-progress? It represents what happed i.e. span was not ended, and does not damage other spans unlike span context leak. WDYT?